extensions/ql-vscode/src/variant-analysis/data-flow-paths-view.ts

```diff
@@ -0,0 +1,62 @@
+import { ExtensionContext, ViewColumn } from "vscode";
+import { AbstractWebview, WebviewPanelConfig } from "../abstract-webview";
+import { assertNever } from "../pure/helpers-pure";
+import { telemetryListener } from "../telemetry";
+import {
+  FromDataFlowPathsMessage,
+  ToDataFlowPathsMessage,
+} from "../pure/interface-types";
+import { DataFlowPaths } from "./shared/data-flow-paths";
+
+export class DataFlowPathsView extends AbstractWebview<
+  ToDataFlowPathsMessage,
+  FromDataFlowPathsMessage
+> {
+  public static readonly viewType = "codeQL.dataFlowPaths";
+
+  public constructor(ctx: ExtensionContext) {
+    super(ctx);
+  }
+
+  public async showDataFlows(dataFlowPaths: DataFlowPaths) {
+    if (!this.isShowingPanel) {
+      await this.getPanel();
+      await this.waitForPanelLoaded();
+    }
+
+    // Focus on panel
+    this.panel?.reveal(undefined, true);
```

Comment:
This seems to be in a different order than the variant analysis view. Does this construct from the variant analysis view not work for this view or should the variant analysis view be changed to use this?

```suggestion
    const panel = await this.getPanel();
    panel.reveal(undefined, true);

    await this.waitForPanelLoaded();
```

Reply:
I think that the use case is just slightly different. With the data flow paths view, we'd want the callers to just say "show the paths" instead of having separate `openView` and `updateView` steps. We also only want to ever allow 1 data flow paths view to ever be open, rather than one for each set of data flow paths it renders.

Reply:
I think that is still possible with the logic from the variant analysis view. The `isShowingPanel` check is implicit when you call `getPanel`, and `waitForPanelLoaded` will just resolve immediately when the panel is already loaded. So, I think the behaviour is the same. I think it would be good to keep these checks consistent between the views.

Reply:
Oh sorry I didn't look at the diff properly. I thought you were suggesting having two functions - `openView` and `editView`. Made the change now.